### PR TITLE
Remove v1.13, Jenkins and Vagrant references

### PR DESCRIPTION
**Pull request by aanm**

### CODE REVIEW

File: .github/renovate.json5 [hunk 1]
```diff
@@ -28,7 +28,6 @@
     "go.mod",
     "go.sum",
     "Makefile.defs",
-    "test/kubernetes-test.sh",
```
Comment (tklauser) [inline .github/renovate.json5:32]: Given this file is removed, we should also remove it from the `fileMatch` section below:

https://github.com/cilium/cilium/blob/5cadeb0e7f4881f64df43fb9a5df30ec6e8dba5b/.github/renovate.json5#L741

File: test/fuzzing/fuzz.go
Comment (joestringer) [inline test/fuzzing/fuzz.go:1]: How is fuzzing related with vagrant?

I would expect these run as regular Go tests, nothing to do with the other changes in this PR.

File: test/kubernetes-test.sh [hunk 2]
```diff
@@ -1,107 +0,0 @@
...
-#   - Service translation is not yet supported, and the tests rely on Services.
-#   - More info at https://github.com/cilium/cilium/issues/5719
-# - should allow egress access to server in CIDR block and
-# - should ensure an IP overlapping both IPBlock.CIDR and IPBlock.Except is allowed and
-# - should enforce except clause while egress access to server in CIDR block
-#   - TL;DR Cilium does not allow to specify pod CIDRs as part of the policy
-#     because it conflicts with the pod's security identity.
-#   - More info at https://github.com/cilium/cilium/issues/9209
-echo "Running upstream NetworkPolicy tests"
-${HOME}/go/bin/kubetest --provider=local --test \
```
Comment (joestringer) [inline test/kubernetes-test.sh:106]: How are we doing this testing now?

File: test/standalone/microk8s-static-pods.sh
Comment (joestringer) [inline test/standalone/microk8s-static-pods.sh:1]: Not jenkins but :+1: if we want static pods testing we should properly set that up. No point in keeping microk8s-based tests around either.

**Review by joestringer — APPROVED** [review]
Just a few comments around the edges for stuff that's not strictly Jenkins. Otherwise, this is an exciting milestone :)

**Review by christarazi — APPROVED** [review]
:broom:

File: test/standalone/microk8s-static-pods.sh
Comment (aanm) [reply test/standalone/microk8s-static-pods.sh:1]: The logic I used was more "if this script is not being executed from CI then it's effectively broken / deprecated"

File: test/fuzzing/fuzz.go
Comment (aanm) [reply test/fuzzing/fuzz.go:1]: How? To run as tests the file should be named `fuzz_test.go`, no? :thinking:

File: test/kubernetes-test.sh (hunk 2 above)
Comment (aanm) [reply test/kubernetes-test.sh:106]: In here https://github.com/cilium/cilium/blob/b684bd7cadafa71adf17f855bacc0fe00c3264aa/.github/workflows/conformance-k8s-kind-network-policies.yaml#L198

File: test/fuzzing/fuzz.go
Comment (joestringer) [reply test/fuzzing/fuzz.go:1]: Looking around in this directory a bit more I'm pretty sure these are being run by the automated system from oss-fuzz that also reports failures to us via the security email list. I would keep this directory around.

**Review by joestringer — CHANGES_REQUESTED** [review]
Let's keep `test/fuzzing`. The rest looks fine to me.

**Review by michi-covalent — APPROVED** [review]
🧹

**Review by tklauser — APPROVED** [review]

File: .github/renovate.json5 [hunk 3]
```diff
@@ -68,7 +67,7 @@
     "main",
     "v1.16",
     "v1.15",
-    "v1.14",
+    "v1.14"
```
Comment (joestringer) [inline .github/renovate.json5]: nit: json5 allows the existing syntax with a comma at the end of an entry in a list, which is nice as it means we don't have to modify unrelated lines when making changes like this.

Comment (joestringer) [reply .github/renovate.json5]: All similar changes should not be necessary.

**Review by joestringer — APPROVED** [review]

**Review by joamaki — APPROVED** [review]
